File: test/CourseRegistrationService_test.java
import org.junit.jupiter.api.Test;

import java.util.ArrayList;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CourseRegistrationService_test {
    @Test
    public void checkAccesiableRequests_test() {
        DataManagement dataManagement = new DataManagement();
        Advisor advisor = dataManagement.getAdvisor("hilalozkan@marun.edu.tr");
        Student student = dataManagement.getStudent("leylapala@marun.edu.tr");
        //courses
        CourseRegistrationService courseRegistrationService = new CourseRegistrationService();
        courseRegistrationService.createCourseRequest(student, dataManagement.getCourse("Fluid Mechanics"));
        courseRegistrationService.createCourseRequest(student, dataManagement.getCourse("Heat Transfer"));
        courseRegistrationService.createCourseRequest(student, dataManagement.getCourse("Material Sci."));
        ArrayList<CourseRequest> trueList = new ArrayList<>();
        trueList.add(new CourseRequest(student, dataManagement.getCourse("Fluid Mechanics"), student.get_advisorID()));
        trueList.add(new CourseRequest(student, dataManagement.getCourse("Heat Transfer"), student.get_advisorID()));
        trueList.add(new CourseRequest(student, dataManagement.getCourse("Material Sci."), student.get_advisorID()));

        ArrayList<CourseRequest> calculatedList = courseRegistrationService.checkAccesiableRequests(advisor);
        for (int i = 0; i < calculatedList.size(); i++) {
            assertEquals(trueList.get(i).get_course().getCourseName(), calculatedList.get(i).get_course().getCourseName());
        }
    }
}

File: Test/CourseRegistrationService_test.py
import unittest

from pythonCode.CourseRegistrationService import CourseRegistrationService, CourseRequest
from pythonCode.DataManagement import DataManagement


class CourseRegistrationService_test(unittest.TestCase):
    def test_checkAccesiableRequests_test(self):
        data_management = DataManagement()
        advisor = data_management.getAdvisor("sevdaucar@marun.edu.tr")
        student = data_management.getStudent("selinpolat@marun.edu.tr")

        courseRegistrationService = CourseRegistrationService()
        courseRegistrationService.createCourseRequest(student,data_management.getCourse("Fluid Mechanics"))
        courseRegistrationService.createCourseRequest(student,data_management.getCourse("Heat Transfer"))
        courseRegistrationService.createCourseRequest(student,data_management.getCourse("Material Sci."))
        trueList = []
        trueList.append(CourseRequest(student,data_management.getCourse("Fluid Mechanics"),student.get_advisorID()))
        trueList.append(CourseRequest(student,data_management.getCourse("Heat Transfer"),student.get_advisorID()))
        trueList.append(CourseRequest(student,data_management.getCourse("Material Sci."),student.get_advisorID()))

        calculatedList = courseRegistrationService.checkAccesiableRequests(advisor)
        for i in range(len(trueList)):
            self.assertEqual(trueList[i].get_course().getCourseName(), calculatedList[i].get_course().getCourseName(), "Requests are not calculated correctly!")
